balboa-kafka/src/main/scala/com/socrata/balboa/kafka/consumer/Consumer.scala

```diff
@@ -0,0 +1,74 @@
+package com.socrata.balboa.kafka.consumer
+
+import java.io.IOException
+
+import com.socrata.balboa.metrics.data.DataStore
+import com.socrata.balboa.metrics.impl.JsonMessage
+import kafka.consumer.{ConsumerIterator, KafkaStream}
+
+/**
+ * Created by Michael Hotan on 2/18/15.
+ *
+ * Balboa Kafka Consumer
+ */
+sealed abstract class BalboaConsumer(stream: KafkaStream[Array[Byte], Array[Byte]]) extends Runnable {
+
+  override def run(): Unit = {
+    val it: ConsumerIterator[Array[Byte], Array[Byte]] = stream.iterator()
+    while (it.hasNext())
```

Comment:
Looking into it.  It looks like ActiveMQReceiver in balboa-jms is utilizing ActiveMQSession rollback functionality.  I see how BalboaFastFailCheck is being utilized in Cassandra11QueryImpl though.  "Should" be straight forward implementation.